projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts

```diff
@@ -350,335 +243,51 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
             });
     }
 
-    private createCategoryLabels(): Array<string> {
-        if (!this.ebvTree || this.ebvTree.tree.groups.length === 0) {
-            return [];
-        }
-
-        let hierarchyDepth = 1;
-        let group = this.ebvTree.tree.groups[0];
-
-        while (group.groups.length > 0) {
-            hierarchyDepth++;
-            group = group.groups[0];
-        }
+    layerSelected(id?: ProviderLayerIdDict): void {
```

Comment:
Here, you request the id from the portal in any case.

But if there is already something selected via id, you don't select the new id.

I guess we should either set the new id or not request data for the new id.

Reply:
I don't understand what you mean. This method is the callback for the selection of a layer in the dropdown component. It doesn't request anything from the portal.

Reply:
It does, from the portal API. Just change the `id=…` in the URL bar.

Reply:
the queryParameter observable says pipe first, so how can anything happen if you change it at all?